libmamba/src/core/query.cpp

```diff
@@ -275,29 +270,26 @@ namespace mamba
 
     query_result Query::depends(const std::string& query, bool tree) const
     {
-        MQueue job, solvables;
+        solv::ObjQueue job, solvables;
 
-        Id id = pool_conda_matchspec(m_pool.get(), query.c_str());
-        if (id)
-        {
-            job.push(SOLVER_SOLVABLE_PROVIDES, id);
-        }
-        else
+        Id const id = pool_conda_matchspec(m_pool.get(), query.c_str());
+        if (!id)
         {
             throw std::runtime_error("Could not generate query for " + query);
         }
+        job.push_back(SOLVER_SOLVABLE_PROVIDES, id);
 
         query_result::dependency_graph g;
-        selection_solvables(m_pool.get(), job, solvables);
+        selection_solvables(m_pool.get(), job.get(), solvables.get());
 
         int depth = tree ? -1 : 1;
 
-        auto find_latest = [&](MQueue& solvables) -> Solvable*
+        auto find_latest_in_non_empty = [&](solv::ObjQueue& solvables) -> Solvable*
         {
-            Solvable* latest = pool_id2solvable(m_pool.get(), solvables[0]);
-            for (int i = 1; i < solvables.count(); ++i)
+            Solvable* latest = pool_id2solvable(m_pool.get(), solvables.front());
+            for (Id const solv : solvables)
             {
```

Comment:
the previous loop started at 1 - is this still correct?

Reply:
Yes, this is like a `min` function where the running `min` is initialized to the first element.
You don't need to compare it again with itself but it does not change either and make for a simpler use of range-based for loop.